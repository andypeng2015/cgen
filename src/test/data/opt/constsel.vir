module foo

export function w $foo(w %x) {
@start:
  %c = eq.w %x, 1_w
  br %c, @sel, @done
@sel:
  %x = sel.w %c, 42_w, 37_w
  jmp @done
@done:
  ret %x
}
