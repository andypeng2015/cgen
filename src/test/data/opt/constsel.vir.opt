module foo

export function w $foo(w %x) {
@2:
  %1 = eq.w %x, 0x1_w ; @8
  %2 = sel.w %1, 0x2a_w, %x ; @9
  ret %2
}
