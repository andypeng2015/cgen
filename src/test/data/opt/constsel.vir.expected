module foo

export function w $foo(w %x) {
@3:
  %0 = eq.w %x, 0x1_w ; @5
  br %0, @0(0x2a_w), @0(%x)
@0(%x.2):
  ret %x.2
}
